elevation_layer/src/elevation_layer.cpp

```diff
@@ -0,0 +1,242 @@
+//
+// Created by eugenio on 16.10.18.
+//
+
+#include <elevation_layer.h>
+#include <pluginlib/class_list_macros.h>
+
+PLUGINLIB_EXPORT_CLASS(elevation_layer::ElevationLayer, costmap_2d::Layer)
+
+using costmap_2d::NO_INFORMATION;
+using costmap_2d::LETHAL_OBSTACLE;
+using costmap_2d::FREE_SPACE;
```

Comment:
I think that we soon need more than binary traversability information. Imagine you have two paths to reach the goal, one path on flat terrain and one path with ramps/steps etc. Both paths are traversable, but one has higher costs. In these cases the planner will need get these costs through the map. Therefore we will need traversability information between 0 and 1.

Unrelated to that, during testing we saw that noise could result to untraversable cells in the costmap. In these cases the planner failed to find a path. In my opinion it would make sense to have a threshold below which untraversable cells are not considered to be an obstacle.

Reply:
About the first point, yes its true, this one is the short term solution, in the future we will need a non-binary cost. About the second one, it is already implemented like this :)